HTMLTestRunner.py

```diff
@@ -668,9 +668,10 @@ def _generate_report(self, result):
                 desc = desc,
                 count = np+ns+nf+ne,
                 Pass = np,
+                skip = ns,
                 fail = nf,
                 error = ne,
-                cid = 'c%s' % (cid+1),
```

Comment:
As far as the rest of the commit, it looks fine to me.

As to a test case, I am slammed with work and am working a lot of overtime. I am not sure I can carve out much time right now. I am happy to review whatever is submitted, but I can't make promises as to actually generating any code myself.